ui/src/app/utils/resources.tsx

```diff
@@ -0,0 +1,95 @@
+const isNotebook = (resource: WorkspaceResource): boolean => !!resource.notebook;
+
+function toDisplay(resourceType: ResourceType): string {
+  return fp.cond([
+      [rt => rt === ResourceType.COHORT, () => 'Cohort'],
+      [rt => rt === ResourceType.COHORTREVIEW, () => 'Cohort Review'],
+      [rt => rt === ResourceType.CONCEPTSET, () => 'Concept Set'],
+      [rt => rt === ResourceType.DATASET, () => 'Dataset'],
+      [rt => rt === ResourceType.NOTEBOOK, () => 'Notebook'],
+
+      [rt => rt === ResourceType.COHORTSEARCHGROUP, () => 'Group'],
+      [rt => rt === ResourceType.COHORTSEARCHITEM, () => 'Item'],
+      [rt => rt === ResourceType.WORKSPACE, () => 'Workspace'],
+  ])(resourceType);
+}
+
+const getTypeString = (resource: WorkspaceResource): string => toDisplay(getType(resource));
+
+function getDescription(resource: WorkspaceResource): string {
+  return fp.cond([
+      [isCohort, r => r.cohort.description],
+      [isCohortReview, r => r.cohortReview.description],
+      [isConceptSet, r => r.conceptSet.description],
+      [isDataSet, r => r.dataSet.description],
+      [isNotebook, fp.stubString /* notebooks don't have descriptions */]
+  ])(resource);
+}
+
+function getDisplayName(resource: WorkspaceResource): string {
+  return fp.cond([
+      [isCohort, r => r.cohort.name],
+      [isCohortReview, r => r.cohortReview.cohortName],
+      [isConceptSet, r => r.conceptSet.name],
+      [isDataSet, r => r.dataSet.name],
+      [isNotebook, r => dropNotebookFileSuffix(r.notebook.name)],
+  ])(resource);
+}
+
+function getId(resource: WorkspaceResource): number {
+  // Notebooks do not have IDs
+  return fp.cond([
+      [isCohort, r => r.cohort.id],
+      [isCohortReview, r => r.cohortReview.cohortReviewId],
+      [isConceptSet, r => r.conceptSet.id],
+      [isDataSet, r => r.dataSet.id],
+  ])(resource);
+}
+
+function getResourceUrl(resource: WorkspaceResource): string {
+  const {workspaceNamespace, workspaceFirecloudName} = resource;
+  const workspacePrefix = `/workspaces/${workspaceNamespace}/${workspaceFirecloudName}`;
+
+  return fp.cond([
+      [isCohort, r => `${workspacePrefix}/data/cohorts/build?cohortId=${r.cohort.id}`],
+      [isCohortReview, r => `${workspacePrefix}/data/cohorts/${r.cohortReview.cohortId}/review`],
+      [isConceptSet, r => `${workspacePrefix}/data/concepts/sets/${r.conceptSet.id}`],
+      [isDataSet, r => `${workspacePrefix}/data/data-sets/${r.dataSet.id}`],
+      [isNotebook, r => `${workspacePrefix}/notebooks/preview/${encodeURIComponentStrict(r.notebook.name)}`],
+  ])(resource);
+}
+
+function getType(resource: WorkspaceResource): ResourceType {
+  return fp.cond([
+      [isCohort, () => ResourceType.COHORT],
+      [isCohortReview, () => ResourceType.COHORTREVIEW],
+      [isConceptSet, () => ResourceType.CONCEPTSET],
+      [isDataSet, () => ResourceType.DATASET],
+      [isNotebook, () => ResourceType.NOTEBOOK],
+  ])(resource);
+}
+
+export {
```

Comment:
can we also move get-resource-card in here? Since it seems like the place to place related functions.

Reply:
That's what I had originally, but it makes a circular reference so I had to split it out.